src/components/TreeViewFinder/TreeViewFinder.tsx

```diff
@@ -258,27 +328,35 @@ const TreeViewFinder = (props) => {
         if (validationButtonText) {
             return validationButtonText;
         } else {
-            const buttonLabelId =
-                selectedProp?.length > 0
+            let buttonLabelId = '';
+            if (Array.isArray(selectedProp)) {
+                buttonLabelId =
+                    selectedProp?.length > 0
+                        ? 'treeview_finder/replaceElementsValidation'
+                        : 'treeview_finder/addElementsValidation';
+            } else {
+                buttonLabelId = selectedProp
                     ? 'treeview_finder/replaceElementsValidation'
                     : 'treeview_finder/addElementsValidation';
+            }
```

Comment:
```suggestion
          buttonLabelId = (Array.isArray(selectedProp) ? selectedProp.length > 0 : selectedProp)
    ? 'treeview_finder/replaceElementsValidation'
    : 'treeview_finder/addElementsValidation';
```

Reply:
Why not but I'm not a fan to use a double ternary operator.